src/pages/login.tsx

```diff
@@ -1,17 +1,41 @@
 import { login, loginUrl } from '@/services/use-auth';
+import axios from 'axios';
 import type { NextPage } from 'next';
 import Link from 'next/link';
-import { FormEvent } from 'react';
+import { useForm } from 'react-hook-form';
+import toast from 'react-hot-toast';
 import useSWRMutation from 'swr/mutation';
 
+type data = {
+  username: string;
+  password: string;
+};
+
 const Login: NextPage = () => {
+  const {
+    register,
+    setValue,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<data>();
+
   const { trigger } = useSWRMutation(loginUrl, login);
 
-  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
-    event.preventDefault();
-    const data = new FormData(event.currentTarget);
-    trigger({ formData: data });
-  };
+  const onSubmit = handleSubmit(async ({ username, password }) => {
+    try {
+      let formData = new FormData();
+      formData.append('username', username);
+      formData.append('password', password);
+
+      await trigger({ formData });
+    } catch (error: unknown) {
+      let errorMessage = 'Oops, something went wrong!';
+      if (axios.isAxiosError(error)) {
+        errorMessage = error?.response?.data.detail;
+      }
+      toast.error(errorMessage);
```

Comment:
The request error will be shown by a toast while the validation errors will be displayed right in the form component